### ISSUE TITLE
ernie-doc的文本分类效果不理想

### ISSUE THREAD

**Issue by Phoebe0126** [PaddlePaddle/ERNIE#674]
具体参数配置如下：
batch_size: 16
checkpoints: ./output
decr_every_n_nan_or_inf: 2
decr_ratio: 0.8
dev_set: ./data/THUCNews/dev.txt
do_lower_case: True
do_test: False
do_train: True
do_val: True
doc_stride: 128
epoch: 5
ernie_config_path: ./configs/base/zh/ernie_config.json
eval_all: False
for_cn: True
in_tokens: False
incr_every_n_steps: 100
incr_ratio: 5.0
init_checkpoint: 
init_loss_scaling: 1.0
init_pretraining_params: 
is_distributed: False
is_zh: True
label_map_config: 
layer_decay_ratio: 0.8
learning_rate: 0.00015
lr_scheduler: linear_warmup_decay
max_answer_length: 100
max_query_length: 64
max_seq_len: 100
metrics: True
n_best_size: 20
num_iteration_per_drop_scope: 10
num_labels: 10
random_seed: 1
rel_pos_params_sharing: False
repeat_input: False
save_steps: 10000
skip_steps: 10
stream_job: None
test_set: ./data/THUCNews/test.txt
tokenizer: FullTokenizer
train_all: False
train_set: ./data/THUCNews/train.txt
use_amp: False
use_cuda: True
use_dynamic_loss_scaling: False
use_fast_executor: True
use_recompute: False
use_vars: False
validation_steps: 100
verbose: False
vocab_path: ./configs/base/zh/vocab.txt
warmup_proportion: 0.1
weight_decay: 0.01
weight_sharing: True

finetuning start
attention_probs_dropout_prob: 0.1
epsilon: 1e-12
hidden_act: gelu
hidden_dropout_prob: 0.1
hidden_size: 768
initializer_range: 0.02
max_position_embeddings: 512
memory_len: 128
num_attention_heads: 12
num_hidden_layers: 12
sent_type_vocab_size: 4
task_type_vocab_size: 3
vocab_size: 28000

args.is_distributed: False
Device count 4, trainer_id:0
args.vocab_path ./configs/base/zh/vocab.txt
Device count: 4, trainer_id: 0
Num train examples: 180000
Max train steps: 14062
Num warmup steps: 1406

W0523 22:12:34.754221 14073 device_context.cc:404] Please NOTE: device: 0, GPU Compute Capability: 6.1, Driver API Version: 10.2, Runtime API Version: 10.2
W0523 22:12:34.758791 14073 device_context.cc:422] device: 0, cuDNN Version: 7.6.
train pyreader queue size: 70, 
epoch: 0, worker_index: 0, progress: 5248/180000, step: 10, ave loss: 2.280549, ave acc: 0.176471, time cost: 4.076458, speed: 2.453110 steps/s, learning_rate: 0.000001
train pyreader queue size: 70, 
epoch: 0, worker_index: 0, progress: 5888/180000, step: 20, ave loss: 2.332018, ave acc: 0.000000, time cost: 3.281870, speed: 3.047043 steps/s, learning_rate: 0.000002
train pyreader queue size: 70, 
epoch: 0, worker_index: 0, progress: 6528/180000, step: 30, ave loss: 2.327189, ave acc: 0.058824, time cost: 3.284722, speed: 3.044398 steps/s, learning_rate: 0.000003
train pyreader queue size: 70, 
epoch: 0, worker_index: 0, progress: 7168/180000, step: 40, ave loss: 2.282611, ave acc: 0.117647, time cost: 3.292336, speed: 3.037358 steps/s, learning_rate: 0.000004
train pyreader queue size: 70, 
epoch: 0, worker_index: 0, progress: 7808/180000, step: 50, ave loss: 2.297273, ave acc: 0.117647, time cost: 3.289208, speed: 3.040246 steps/s, learning_rate: 0.000005
train pyreader queue size: 70, 
epoch: 0, worker_index: 0, progress: 8448/180000, step: 60, ave loss: 2.354424, ave acc: 0.058824, time cost: 3.298677, speed: 3.031518 steps/s, learning_rate: 0.000006
train pyreader queue size: 70, 
epoch: 0, worker_index: 0, progress: 9088/180000, step: 70, ave loss: 2.271231, ave acc: 0.117647, time cost: 3.301912, speed: 3.028548 steps/s, learning_rate: 0.000007
train pyreader queue size: 70, 
epoch: 0, worker_index: 0, progress: 9728/180000, step: 80, ave loss: 2.345518, ave acc: 0.058824, time cost: 3.331487, speed: 3.001663 steps/s, learning_rate: 0.000008
train pyreader queue size: 70, 
epoch: 0, worker_index: 0, progress: 10368/180000, step: 90, ave loss: 2.306730, ave acc: 0.000000, time cost: 3.303587, speed: 3.027012 steps/s, learning_rate: 0.000009
train pyreader queue size: 70, 
epoch: 0, worker_index: 0, progress: 11008/180000, step: 100, ave loss: 2.327241, ave acc: 0.058824, time cost: 3.320735, speed: 3.011382 steps/s, learning_rate: 0.000011
processining example: 1000
processining example: 2000
data num: 2512
[100_eval evaluation] ave loss: 2.302494, ave acc: 0.104299, data_num: 2512, elapsed time: 19.938949 s

**Comment by dingsiyu** [comment]
您好，
（1）根据您贴出来的log记录，只有step_100的结果，模型仍处于欠拟合状态。根据我们自己的finetune实验的log记录，THUCNews的step_100评估结果与您的结果是可比的状态，您可以再继续观察后续的fientune结果，如果仍然不符合预期，可再联系我们。
（2）此外，根据您的log记录，dev_set的数据为2512条，而我们使用的版本，dev_set是5000条数据，您可以再check下所使用的数据。（数据参考地址：https://github.com/gaussic/text-classification-cnn-rnn）
（3）基于上述版本的THUCNews数据，在finetune参数配置中，epoch=16（具体可参考论文附录）

**Comment by Phoebe0126** [comment]
> 您好，
> （1）根据您贴出来的log记录，只有step_100的结果，模型仍处于欠拟合状态。根据我们自己的finetune实验的log记录，THUCNews的step_100评估结果与您的结果是可比的状态，您可以再继续观察后续的fientune结果，如果仍然不符合预期，可再联系我们。
> （2）此外，根据您的log记录，dev_set的数据为2512条，而我们使用的版本，dev_set是5000条数据，您可以再check下所使用的数据。（数据参考地址：[https://github.com/gaussic/text-classification-cnn-rnn）](https://github.com/gaussic/text-classification-cnn-rnn%EF%BC%89)
> （3）基于上述版本的THUCNews数据，在finetune参数配置中，epoch=16（具体可参考论文附录）

您好！感谢您的解答，之前的问题得到了解决。另外我还想询问一下关于文本分类的标签处理问题。对于不同的chunk，如果得到的label_for_one_sample标签不完全相同，这时候该长文本的分类结果应如何处理呢？
